vaadin-ordered-layout-flow/src/main/java/com/vaadin/flow/component/orderedlayout/ScrollContainer.java

```diff
@@ -0,0 +1,174 @@
+     * {@link #setScrollDirection(ScrollDirection)}.
+     * 
+     */
+    public ScrollContainer() {
+    }
+
+    /**
+     * Convenience constructor to create a container with the given content.
+     * Scroll direction defaults to {@link ScrollDirection#BOTH} and can be set
+     * with {@link #setScrollDirection(ScrollDirection)}.
+     * 
+     * @param content
+     *            the content of this container
+     * @see #setContent(Component)
+     */
+    public ScrollContainer(Component content) {
+        this();
+        setContent(content);
+    }
+
+    /**
+     * Convenience constructor to create a container with the given content and
+     * scroll direction.
+     *
+     * @param content
+     *            the content of this container
+     * @param scrollDirection
+     *            scroll direction that the container will have
+     */
+    public ScrollContainer(Component content, ScrollDirection scrollDirection) {
+        this();
+        setContent(content);
+        setScrollDirection(scrollDirection);
+    }
+
+    /**
+     * Convenience constructor to create an empty container with the given
+     * scroll direction. Content can be set with {@link #setContent(Component)}
+     *
+     * @param scrollDirection
+     *            scroll direction that the container will have
+     */
+    public ScrollContainer(ScrollDirection scrollDirection) {
+        this();
+        setScrollDirection(scrollDirection);
+    }
+
+    /**
+     * Sets the content of this container.
+     *
+     * The content must always be set, either with a constructor parameter or by
+     * calling this method.
+     * 
+     * @param content
+     *            a component to use as content
+     */
+    public void setContent(Component content) {
+        if (this.content != null) {
+            this.content.getElement().removeFromParent();
+        }
+        this.content = content;
+        this.getElement().appendChild(content.getElement());
+    }
+
+    /**
+     * Gets the content of this container.
+     *
+     * @return the component used as content
+     */
+    public Component getContent() {
+        return content;
+    }
+
+    /**
+     * Sets the scroll direction for this container. Defaults to
+     * {@link ScrollDirection#BOTH}.
+     *
+     * @param scrollDirection
+     *            {@link ScrollDirection#BOTH} to enable both vertical and
+     *            horizontal scrollbars. {@link ScrollDirection#HORIZONTAL} to
+     *            enable only horizontal scrollbars.
+     *            {@link ScrollDirection#VERTICAL} to enable only vertical
+     *            scrollbars. {@link ScrollDirection#NONE} to disable both
+     *            vertical and horizontal scrollbars.
+     */
+    public void setScrollDirection(ScrollDirection scrollDirection) {
+        requireNonNull(scrollDirection, "Scroll direction must not be null");
+        if (scrollDirection == ScrollDirection.BOTH) {
```

Comment:
The reason for this `if` is that the second parameter of setAttribute can not be null. I think it would be more clear to test explicitly for that. It would make this less tight coupled with the logic of `scrollDirection.toWebComponentValue`.
```
String webComponentValue = scrollDirection.toWebComponentValue();
if(webComponentValue == null)
...
```
Not blocking though

Reply:
Makes perfect sense, I agree it helps making it less coupled and easier to follow.
I changed it.